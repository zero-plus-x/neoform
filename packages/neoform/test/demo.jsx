import React from 'react';
import { mount } from 'enzyme';

import Demo from '../demo';

describe('neoform', () => {
  it('basic wrapper', () => {
    const wrapper = mount(
      <Demo/>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('change field', () => {
    const wrapper = mount(
      <Demo/>
    );

    const firstName = wrapper.find('[name="firstName"]');

    firstName.simulate('change', {
      target: {
        value: 'hey from tests'
      }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
