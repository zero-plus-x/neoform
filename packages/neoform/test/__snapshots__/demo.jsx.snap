// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`neoform basic wrapper 1`] = `
<Demo>
  <withContext(Form)
    data={
      Object {
        "firstName": "John",
        "friends": Array [
          Object {
            "firstName": "Sad",
            "lastName": "Pepe",
          },
          Object {
            "firstName": "Forever",
            "lastName": "Alone",
          },
        ],
        "lastName": "Doe",
        "phoneNumbers": Array [
          123123,
          456456,
        ],
      }
    }
    getValue={[Function]}
    onChange={[Function]}
  >
    <MyForm
      data={
        Object {
          "firstName": "John",
          "friends": Array [
            Object {
              "firstName": "Sad",
              "lastName": "Pepe",
            },
            Object {
              "firstName": "Forever",
              "lastName": "Alone",
            },
          ],
          "lastName": "Doe",
          "phoneNumbers": Array [
            123123,
            456456,
          ],
        }
      }
      getValue={[Function]}
      onChange={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <h1>
          simple form
        </h1>
        <h2>
          personal data
        </h2>
        <div>
          <label>
            first name
            <getContext(withProps(omitProps(Field)))
              name="firstName"
            >
              <withProps(omitProps(Field))
                name="firstName"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "John",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="firstName"
                  onChange={[Function]}
                  value="John"
                >
                  <input
                    name="firstName"
                    onChange={[Function]}
                    value="John"
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </label>
        </div>
        <div>
          <label>
            last name
            <getContext(withProps(omitProps(Field)))
              name="lastName"
            >
              <withProps(omitProps(Field))
                name="lastName"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "John",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="lastName"
                  onChange={[Function]}
                  value="Doe"
                >
                  <input
                    name="lastName"
                    onChange={[Function]}
                    value="Doe"
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </label>
        </div>
        <h2>
          phone numbers
        </h2>
        <ul>
          <li>
            <getContext(withProps(omitProps(Field)))
              name="phoneNumbers[0]"
            >
              <withProps(omitProps(Field))
                name="phoneNumbers[0]"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "John",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="phoneNumbers[0]"
                  onChange={[Function]}
                  value={123123}
                >
                  <input
                    name="phoneNumbers[0]"
                    onChange={[Function]}
                    value={123123}
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </li>
          <li>
            <getContext(withProps(omitProps(Field)))
              name="phoneNumbers[1]"
            >
              <withProps(omitProps(Field))
                name="phoneNumbers[1]"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "John",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="phoneNumbers[1]"
                  onChange={[Function]}
                  value={456456}
                >
                  <input
                    name="phoneNumbers[1]"
                    onChange={[Function]}
                    value={456456}
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </li>
        </ul>
        <h2>
          friends
        </h2>
        <ul>
          <li>
            <div>
              <label>
                first name
                <getContext(withProps(omitProps(Field)))
                  name="friends[0].firstName"
                >
                  <withProps(omitProps(Field))
                    name="friends[0].firstName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "John",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[0].firstName"
                      onChange={[Function]}
                      value="Sad"
                    >
                      <input
                        name="friends[0].firstName"
                        onChange={[Function]}
                        value="Sad"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
            <div>
              <label>
                last name
                <getContext(withProps(omitProps(Field)))
                  name="friends[0].lastName"
                >
                  <withProps(omitProps(Field))
                    name="friends[0].lastName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "John",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[0].lastName"
                      onChange={[Function]}
                      value="Pepe"
                    >
                      <input
                        name="friends[0].lastName"
                        onChange={[Function]}
                        value="Pepe"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
          </li>
          <li>
            <div>
              <label>
                first name
                <getContext(withProps(omitProps(Field)))
                  name="friends[1].firstName"
                >
                  <withProps(omitProps(Field))
                    name="friends[1].firstName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "John",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[1].firstName"
                      onChange={[Function]}
                      value="Forever"
                    >
                      <input
                        name="friends[1].firstName"
                        onChange={[Function]}
                        value="Forever"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
            <div>
              <label>
                last name
                <getContext(withProps(omitProps(Field)))
                  name="friends[1].lastName"
                >
                  <withProps(omitProps(Field))
                    name="friends[1].lastName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "John",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[1].lastName"
                      onChange={[Function]}
                      value="Alone"
                    >
                      <input
                        name="friends[1].lastName"
                        onChange={[Function]}
                        value="Alone"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
          </li>
        </ul>
        <button
          type="submit"
        >
          submit
        </button>
      </form>
    </MyForm>
  </withContext(Form)>
</Demo>
`;

exports[`neoform change field 1`] = `
<Demo>
  <withContext(Form)
    data={
      Object {
        "firstName": "hey from tests",
        "friends": Array [
          Object {
            "firstName": "Sad",
            "lastName": "Pepe",
          },
          Object {
            "firstName": "Forever",
            "lastName": "Alone",
          },
        ],
        "lastName": "Doe",
        "phoneNumbers": Array [
          123123,
          456456,
        ],
      }
    }
    getValue={[Function]}
    onChange={[Function]}
  >
    <MyForm
      data={
        Object {
          "firstName": "hey from tests",
          "friends": Array [
            Object {
              "firstName": "Sad",
              "lastName": "Pepe",
            },
            Object {
              "firstName": "Forever",
              "lastName": "Alone",
            },
          ],
          "lastName": "Doe",
          "phoneNumbers": Array [
            123123,
            456456,
          ],
        }
      }
      getValue={[Function]}
      onChange={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <h1>
          simple form
        </h1>
        <h2>
          personal data
        </h2>
        <div>
          <label>
            first name
            <getContext(withProps(omitProps(Field)))
              name="firstName"
            >
              <withProps(omitProps(Field))
                name="firstName"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "hey from tests",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="firstName"
                  onChange={[Function]}
                  value="hey from tests"
                >
                  <input
                    name="firstName"
                    onChange={[Function]}
                    value="hey from tests"
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </label>
        </div>
        <div>
          <label>
            last name
            <getContext(withProps(omitProps(Field)))
              name="lastName"
            >
              <withProps(omitProps(Field))
                name="lastName"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "hey from tests",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="lastName"
                  onChange={[Function]}
                  value="Doe"
                >
                  <input
                    name="lastName"
                    onChange={[Function]}
                    value="Doe"
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </label>
        </div>
        <h2>
          phone numbers
        </h2>
        <ul>
          <li>
            <getContext(withProps(omitProps(Field)))
              name="phoneNumbers[0]"
            >
              <withProps(omitProps(Field))
                name="phoneNumbers[0]"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "hey from tests",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="phoneNumbers[0]"
                  onChange={[Function]}
                  value={123123}
                >
                  <input
                    name="phoneNumbers[0]"
                    onChange={[Function]}
                    value={123123}
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </li>
          <li>
            <getContext(withProps(omitProps(Field)))
              name="phoneNumbers[1]"
            >
              <withProps(omitProps(Field))
                name="phoneNumbers[1]"
                neoform={
                  Object {
                    "getValue": [Function],
                    "state": Object {
                      "firstName": "hey from tests",
                      "friends": Array [
                        Object {
                          "firstName": "Sad",
                          "lastName": "Pepe",
                        },
                        Object {
                          "firstName": "Forever",
                          "lastName": "Alone",
                        },
                      ],
                      "lastName": "Doe",
                      "phoneNumbers": Array [
                        123123,
                        456456,
                      ],
                    },
                    "updateData": [Function],
                  }
                }
              >
                <MyInput
                  name="phoneNumbers[1]"
                  onChange={[Function]}
                  value={456456}
                >
                  <input
                    name="phoneNumbers[1]"
                    onChange={[Function]}
                    value={456456}
                  />
                </MyInput>
              </withProps(omitProps(Field))>
            </getContext(withProps(omitProps(Field)))>
          </li>
        </ul>
        <h2>
          friends
        </h2>
        <ul>
          <li>
            <div>
              <label>
                first name
                <getContext(withProps(omitProps(Field)))
                  name="friends[0].firstName"
                >
                  <withProps(omitProps(Field))
                    name="friends[0].firstName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "hey from tests",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[0].firstName"
                      onChange={[Function]}
                      value="Sad"
                    >
                      <input
                        name="friends[0].firstName"
                        onChange={[Function]}
                        value="Sad"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
            <div>
              <label>
                last name
                <getContext(withProps(omitProps(Field)))
                  name="friends[0].lastName"
                >
                  <withProps(omitProps(Field))
                    name="friends[0].lastName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "hey from tests",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[0].lastName"
                      onChange={[Function]}
                      value="Pepe"
                    >
                      <input
                        name="friends[0].lastName"
                        onChange={[Function]}
                        value="Pepe"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
          </li>
          <li>
            <div>
              <label>
                first name
                <getContext(withProps(omitProps(Field)))
                  name="friends[1].firstName"
                >
                  <withProps(omitProps(Field))
                    name="friends[1].firstName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "hey from tests",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[1].firstName"
                      onChange={[Function]}
                      value="Forever"
                    >
                      <input
                        name="friends[1].firstName"
                        onChange={[Function]}
                        value="Forever"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
            <div>
              <label>
                last name
                <getContext(withProps(omitProps(Field)))
                  name="friends[1].lastName"
                >
                  <withProps(omitProps(Field))
                    name="friends[1].lastName"
                    neoform={
                      Object {
                        "getValue": [Function],
                        "state": Object {
                          "firstName": "hey from tests",
                          "friends": Array [
                            Object {
                              "firstName": "Sad",
                              "lastName": "Pepe",
                            },
                            Object {
                              "firstName": "Forever",
                              "lastName": "Alone",
                            },
                          ],
                          "lastName": "Doe",
                          "phoneNumbers": Array [
                            123123,
                            456456,
                          ],
                        },
                        "updateData": [Function],
                      }
                    }
                  >
                    <MyInput
                      name="friends[1].lastName"
                      onChange={[Function]}
                      value="Alone"
                    >
                      <input
                        name="friends[1].lastName"
                        onChange={[Function]}
                        value="Alone"
                      />
                    </MyInput>
                  </withProps(omitProps(Field))>
                </getContext(withProps(omitProps(Field)))>
              </label>
            </div>
          </li>
        </ul>
        <button
          type="submit"
        >
          submit
        </button>
      </form>
    </MyForm>
  </withContext(Form)>
</Demo>
`;
